migrations/V203__MTO_legacy_data_migration.sql

```diff
@@ -0,0 +1,250 @@
+/**
+This migration migrates operational solution data to mto solution data. It does the following:
+1. looks at all operational needs currently set to needed for a model plan. For each operational need, a relevant mto milestone is created.
+   a. A matching possible operational need results in a milestone with a matching common milestone
+   b. A custom operational need results in a custom mto milestones
+2. It then looks at all operational solutions for those operational needs. It will attempt to standardize the mto solutions that are inserted. 
+    a. If a solution is duplicated, it will add only one record and link it to both of the new mto milestones
+    b. a possible operational solution becomes a common mto solution when possible
+    c. a custom operational solution becomes a custom mto solution
+
+3.  It will then create a link between the mto milestone and the mto solution.
+
+4. Duplicate custom solutions.
+   a. this migration will get the most recent one and insert that. It will ignore another custom solution on that model plan  with the same name.
+
+
+
+The following concepts from IT solutions are not applicable in the models to operations matrix.
+   a. solution sub tasks
+   b. document solution links
+
+
+Note: the newly created milestones are also not categorized, and do not have facilitated by populated
+**/
+
+
+WITH needs AS ( --noqa
```

Comment:
Looked up `--noqa`, is this needed/what is this suppressing?

Reply:
This is suppressing sql-fluff linting, it will complain if you don't use the CTE later on in the script (and sometime it has problems interpreting it correctly). But I think that's a good call out! Most of these are referenced in subsequent CTEs, so the --noqa can be removed (except for inserted milestones)